### PR TITLE
Milestone1: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by brainconnect93**
- [ ] Created a person file and added the class Person with the necessary constructor
- [ ] Created a student file and added the class Student inherited from the person with the necessary constructor
- [ ] Created a teacher file and added the class Teacher inherited from the person with the necessary constructor

> Kindly review and approve

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    @parent_permission = parent_permission
+  end
+  attr_accessor :name, :age
+  attr_reader :id
+
+  private
+
+  def of_age
+    @age >= 18
+  end
```
Comment (DeliceLydia) [inline person.rb:20]: - [x] The only private method is this `of_age` make ` can_use_services?` public, move it before `of_age` method

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Person
+  def initialize(age, name = 'Unkonwn', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+  attr_accessor :name, :age
+  attr_reader :id
```
Comment (DeliceLydia) [inline person.rb]: ## Optional
- [ ] I suggest that you put these attribute before all the methods

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @brainconnect93, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 1 above)
Comment (brainconnect93) [reply person.rb:20]: done as requested

File: person.rb (hunk 2 above)
Comment (brainconnect93) [reply person.rb]: Rectified

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @brainconnect93 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## ❌ Invalid Code Review Request

You have submitted a project, that has failed linter checks. Kindly fix your linter errors and submit again 👍🏽 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by victorokoroji — APPROVED** [review]
Hi @brainconnect93 , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**